clang/lib/AST/ASTImporter.cpp

```diff
@@ -9896,9 +9903,13 @@ Expected<FileID> ASTImporter::Import(FileID FromID, bool IsBuiltin) {
         // FIXME: The filename may be a virtual name that does probably not
         // point to a valid file and we get no Entry here. In this case try with
         // the memory buffer below.
```

Comment:
Did you check that these FIXME's have no connection to the problem?

Reply:
Invalid file could exist in real case (causes no Entry) while `SourceLocation` overflow is normally caused by cumulative effect of multiple (large enough) files imported so I think these two are not connected.

Reply:
Is there a way to reproduce this? By creating some very large dummy files? Just to confirm what the crash is exactly (since I don't see an attached stacktrace)